wagtail/contrib/sitemaps/tests.py

```diff
@@ -48,21 +50,40 @@ def setUp(self):
 
         self.site = Site.objects.get(is_default_site=True)
 
+        # Clear the cache to that runs are deterministic regarding the sql count
+        ContentType.objects.clear_cache()
+
     def test_items(self):
-        sitemap = Sitemap(self.site)
+        request = RequestFactory().get('/sitemap.xml')
+
+        sitemap = Sitemap(self.site, request)
         pages = sitemap.items()
 
         self.assertIn(self.child_page.page_ptr.specific, pages)
         self.assertNotIn(self.unpublished_child_page.page_ptr.specific, pages)
         self.assertNotIn(self.protected_child_page.page_ptr.specific, pages)
 
-    def test_get_urls(self):
+    def test_get_urls_without_request(self):
         request = RequestFactory().get('/sitemap.xml')
         req_protocol = request.scheme
         req_site = get_current_site(request)
 
         sitemap = Sitemap(self.site)
-        urls = [url['location'] for url in sitemap.get_urls(1, req_site, req_protocol)]
+        with self.assertNumQueries(18):
+            urls = [url['location'] for url in sitemap.get_urls(1, req_site, req_protocol)]
+
+        self.assertIn('http://localhost/', urls)  # Homepage
+        self.assertIn('http://localhost/hello-world/', urls)  # Child page
+
+    def test_get_urls_with_request_site_cache(self):
+        request = RequestFactory().get('/sitemap.xml')
+        req_protocol = request.scheme
+        req_site = get_current_site(request)
+
+        sitemap = Sitemap(self.site, request)
+
+        with self.assertNumQueries(16):
+            urls = [url['location'] for url in sitemap.get_urls(1, req_site, req_protocol)]
```

Comment:
In fact, this pull request only removes 2 SQL queries. This number doesn’t seem to increase with the number of pages, probably simply because site fetching uses a cache system.

Of course, with the previous comment (specifying `request` instead of both `request` and `site`), this PR doesn’t add complexity, so these 2 extra queries are still good to save.